tests/test_zlib.rs

```diff
@@ -0,0 +1,331 @@
+        let cmd_opts = vec![String::from("make"),
+                            // "-C".to_string(),
+                            // String::from(my_tgt_path.to_str().unwrap())
+            ];
+        let gen_opts = BitcodeOptions { clang_path: common::user_clang_cmd(),
+                                        bcout_path: None,
+                                        suppress_automatic_debug: false,
+                                        inject_arguments: Vec::new(),
+                                        remove_arguments: Vec::new(),
+                                        verbose: vec![true, true],
+                                        strict: false,
+                                        command: cmd_opts,
+                                        any_fail: false };
+        {
+            let _push2 = pushd(my_tgt_path)?;
+            common::gen_bitcode(gen_opts.clone())?;
+        }
+
+        let mut exe_path = PathBuf::from(my_tgt_path);
+        exe_path.push("example64");
+        assert!(exe_path.exists());  // build successfully generated a static executable
+
+        // Now make specific changes and rebuild *without* build-bom
+        Cmd::new("sed").arg("-ie").arg("s,Copyright,COPYRIGHT,")
+            .arg(my_tgt_path.join("gzread.c"))
+            .run()?;
+        Cmd::new("make").arg("-C").arg(my_tgt_path)
+            .arg("-f").arg("Makefile.in").arg("libz.a").run()?;
+
+        // And finally, build the remaining parts with build-bom
+        {
+            let _push2 = pushd(my_tgt_path)?;
+            common::gen_bitcode(gen_opts)?;
+        }
+        assert!(exe_path.exists());  // build successfully generated a static executable
+
+        let mut bc_path = PathBuf::from(my_tgt_path);
+        bc_path.push("example64.bc");
+        let bc_path2 = bc_path.clone();
+        let extract_opts = ExtractOptions { input: exe_path,
+                                            output: bc_path,
+                                            llvm_link_path: common::user_llvm_link_cmd(),
+                                            verbose: vec![true, true]};
+        common::extract_bitcode(extract_opts)?;
+
+        // Bitcode extracted from an executable linked against a static library
+        // contains the bitcode from the executable and bitcode from every module in
+        // the static library that the executable utilizes.
+
+        let ll = get_llvm_str(&bc_path2)?;
+        let fnd = vec![
+            ll.contains("inflateCopy"),   // static lib
+            ll.contains("compressBound"),   // static lib
+            ll.contains("deflateSetHeader"),   // static lib
+            ll.contains("gzfread"),   // static lib, rebuilt outside of build-bom
+            ll.contains("gzfwrite"),   // static lib
+            ll.contains("test_gzio"),  // exe src
+        ];
+        assert_eq!(fnd, vec![true, true, true, false, true, true]);
+    }
+    assert!(tdir.path().exists()); // ensure tempdir lasts until here
```

Comment:
Is this testing that build-bom didn't delete it? If you are just ensuring that the test harness isn't broken, you could use the `mem::drop` trick here too

Reply:
Actually, this is a static that is shared by all 5 tests, so a particular test cannot/should not force it to be dropped because it will still be used by the others.  The reference at the end is just a bit of additional insurance that no optimization or accidental move caused it to be dropped earlier.